src/Slot-Core/TemporaryVariable.class.st

```diff
@@ -23,19 +23,10 @@ Class {
 
 { #category : #'class initialization' }
 TemporaryVariable class >> initialize [
-	Properties
-		ifNil: [ self initializeProperties ]
-		ifNotNil: [ | newDict | 
-			newDict := WeakIdentityKeyDictionary newFrom: Properties.
-			newDict rehash.
-			Properties := newDict ].
-	PersistentTemps ifNil: [ PersistentTemps := Set new]
-		
-]
-
-{ #category : #'class initialization' }
-TemporaryVariable class >> initializeProperties [
-	Properties := WeakIdentityKeyDictionary new.
+	Properties := Properties
+		ifNil: [ WeakIdentityKeyDictionary new ]
+		ifNotNil: [ (WeakIdentityKeyDictionary newFrom: Properties) rehash ].
+	PersistentTemps ifNil: [ PersistentTemps := Set new ]
```

Comment:
I've seen this code three times already in this PR... wouldn't it be worth it to externalize this maybe?

Reply:
We could do a Trait, but we ca nit use traits in the kernel...